### ISSUE TITLE
Unable to generate latexpdf sphinx document in none standard directory structure, issue in gaphor/extensions/sphinx.py?

### ISSUE THREAD

**Issue by 2old4it** [gaphor/gaphor#1902]
### Describe the bug
First of all I don't know if it is really a bug or if I misuse the way gaphor must be used.
I have gaphor installed under a virtual environment BUT not where my project files are.
If I try to make a pdf document using sphinx I but get an exception:

```
Exception occurred:
  File "/usr/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '_build/gaphor'
```

### To Reproduce
My setup is:
`/venv_path/PENV` is where gaphor is installed

`/project_path/gaphor-model` is where the gaphor model is stored
`/project_path/source` is where the *.rst files are located
`/project_path/source/_build` is where I want the build directory to be
`/project_path` is where I execute the command.

so I do:
- cd /project_path
- source /venv_path/PENV/bin/activate
- sphinx-build -M latexpdf "./source" "./source/_build/"

and than I get:

```
Running Sphinx v5.3.0
making output directory... done
Gaphor models: {'TEST': '../gaphor-model/test.gaphor'}
building [mo]: targets for 0 po files that are out of date
building [latex]: all documents
updating environment: [new config] 7 added, 0 changed, 0 removed
reading sources... [ 14%] concept                                                                                                                                                            
Exception occurred:
  File "/usr/lib/python3.10/pathlib.py", line 1175, in mkdir
    self._accessor.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '_build/gaphor'
The full traceback has been saved in /tmp/sphinx-err-2788eeov.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```

The comment suggests I should report an issue at sphinx-doc but the problem/solution for me is in the gaphor installation.

This "solution" works for me.
/venv_path/PENV/lib/python3.10/site-packages/gaphor/extensions/sphinx.py

```
diff --git a/sphinx_old.py b/sphinx.py
index 189efaa..5e36a99 100644
--- a/sphinx_old.py
+++ b/sphinx.py
@@ -71,7 +71,7 @@ class DiagramDirective(sphinx.util.docutils.SphinxDirective):
         outdir = (
             (Path(self.env.app.doctreedir) / ".." / "gaphor")
             .resolve()
-            .relative_to(self.env.srcdir)
+#2old4it            .relative_to(self.env.srcdir)
         )
         outdir.mkdir(exist_ok=True)
 
@@ -98,6 +98,11 @@ class DiagramDirective(sphinx.util.docutils.SphinxDirective):
         save_svg(outfile.with_suffix(".svg"), diagram)
         save_pdf(outfile.with_suffix(".pdf"), diagram)
 
+        # 2old4it
+        outdir = outdir.relative_to(self.env.srcdir)
+        outfile = outdir / f"{diagram.id}"
+
         # Image needs a relative path. Make our outfile path relative to the doc
         for _ in Path(self.env.docname).parts[:-1]:
             outfile = Path("..") / outfile
```

Then I get a nice generated pdf document including the gaphor models I have defined in the *rst files.
e.g. using:

```
.. diagram:: Technology Structure
   :model: TEST
```

### Expected behavior
Well no exceptions.

### OS
- Linux 5.15.78-1-MANJARO x86_64 GNU/Linux

### Version
Version of Gaphor:
- gaphor 2.13.0
- Sphinx version: 5.3.0
- Python version: 3.10.8 (CPython)
- Docutils version: 0.19 
- Jinja2 version: 3.1.2

**Comment by amolenaar** [comment]
@2old4it  Thanks for the issue report. Your change looks solid. Do you want to make a PR for it, so we can test it on the Gaphor code base as well?

**Comment by danyeaw** [comment]
@all-contributors please add @2old4it for bug

Closed by #1905.
